driver-pulsar/src/main/java/io/openmessaging/benchmark/driver/pulsar/PulsarBenchmarkDriver.java

```diff
@@ -97,11 +91,11 @@ public String getTopicNamePrefix() {
 
     @Override
     public CompletableFuture<BenchmarkConsumer> createConsumer(String topic, String subscriptionName,
-            ConsumerCallback consumerCallback) {
+                                                               ConsumerCallback consumerCallback, int partitionsPerTopic) {
         ConsumerConfiguration conf = new ConsumerConfiguration();
         conf.setSubscriptionType(SubscriptionType.Failover);
         conf.setMessageListener((consumer, msg) -> {
-            consumerCallback.messageReceived(msg.getData());
+            consumerCallback.messageReceived(msg.getData(), System.nanoTime());
```

Comment:
`System.nanoTime()` is not portable across difference machines. We should just use the `publishTimestamp` to calculate the end-to-end latency in milliseconds.
Also, if the consumer is in a different machine, the end-to-end latency measurement can be impacted by the clock skew (even with NTP, different machine can drift from 10s to 100s of millis).

Reply:
Valid point, I'll switch this and `publishTimestamp` to millis.

Reply:
While testing the unit change locally, it wasn't granular enough and I'd get zeros. Generally, if there is time drift and the results get affected by that means there is no problem with consumer performance (or producer for that matter). Whether this is millis or nanos doesn't matter much once you load up the system and the results get "interesting". I'm going to switch it back to nanos for consistency with produce latency metric sake.

Reply:
What I was meaning here is that calling `System.nanoTime()` on 2 different machines will give different values. It's not representing a "relative timestamp". 

From the javadocs :

> The values returned by this method become meaningful only when the difference between two such values, obtained within the same instance of a Java virtual machine, is computed.

Reply:
I'm happy to switch this to millis, but then produce latency would need the same treatment to avoid the other misleading results result I was seeing.